netty/src/main/java/io/grpc/transport/netty/NettyClientHandler.java

```diff
@@ -241,15 +241,19 @@ public boolean visit(Http2Stream stream) throws Http2Exception {
   }
 
   @Override
+  public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
+    if (getEmbeddedHttp2Exception(cause) != null) {
+      super.exceptionCaught(ctx, cause);
+    } else {
+      goAwayStatus(Status.fromThrowable(cause));
```

Comment:
Hmm... Yeah, we should, although I don't think gracefully since we don't know what state we are in.

Best way to do that is `Http2ConnectionHandler.goAway()` with HTTP/2's INTERNAL_ERROR?